### PR TITLE
Sdk new params

### PR DESCRIPTION
**Pull request by RTLcoil**
Added various new params to SDK.

### CODE REVIEW

File: Cloudinary.Test/CloudinaryTest.cs [hunk 1]
```diff
@@ -705,11 +705,12 @@ public void TestPublishByTag()
                 ResourceType = ResourceType.Image
             });
 
-            Assert.AreEqual(publish_result.Published.Count, 1);
+            
 
             DelResResult delResult = m_cloudinary.DeleteResourcesByTag(
                 "TestForPublish");
 
+            Assert.AreEqual(publish_result.Published.Count, 1);
```
Comment (ronen-burin) [inline Cloudinary.Test/CloudinaryTest.cs]: why did you change the order? it's more clear when you check the count right after the operation.

File: Cloudinary.Test/CloudinaryTest.cs [hunk 2]
```diff
@@ -2281,7 +2282,7 @@ public void TestUsage()
             Assert.True(result.Resources > 0);
             Assert.True(result.Objects.Used < result.Objects.Limit);
             Assert.True(result.Bandwidth.Used < result.Bandwidth.Limit);
-            Assert.True(result.Storage.Used < result.Storage.Limit);
```
Comment (ronen-burin) [inline Cloudinary.Test/CloudinaryTest.cs]: why was it removed?

File: Cloudinary.Test/CloudinaryTest.cs [hunk 3]
```diff
@@ -2297,7 +2298,7 @@ public void TestUsageAsync()
...
```
Comment (ronen-burin) [inline Cloudinary.Test/CloudinaryTest.cs]: why was it removed?

File: Cloudinary/Shared/Coudinary.NetCoreShared/Actions/ContextParams.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        /// <summary>
+        /// The action to perform on image resources using the given tag.
+        /// </summary>
+        public ContextCommand Command { get; set; }
+
+        /// <summary>
+        /// Validate object model
+        /// </summary>
+        public override void Check()
```
Comment (ronen-burin) [inline Cloudinary/Shared/Coudinary.NetCoreShared/Actions/ContextParams.cs]: why do we need this method? what is the parent behavior?

Comment (RTLcoil) [inline Cloudinary/Shared/Coudinary.NetCoreShared/Actions/ContextParams.cs]: This method exists in each param class and is used for parameters validation before sending parameters to server. In base class this method checks some basic parameters. This method is always present in parameter class, even if the method does nothing (some param classes do not require validation). It was like this initially and we decided not to change code's style.
